include/ppx/geometry.h

```diff
@@ -59,6 +59,8 @@ struct GeometryOptions
     uint32_t                      vertexBindingCount                      = 0;
     grfx::VertexBinding           vertexBindings[PPX_MAX_VERTEX_BINDINGS] = {};
     grfx::PrimitiveTopology       primtiveTopology                        = grfx::PRIMITIVE_TOPOLOGY_TRIANGLE_LIST;
+    uint32_t                      finalIndexCount                         = 0;
+    uint32_t                      finalVertexCount                        = 0;
 
```

Comment:
Can you help me understand what is meant by final here? Is the intention that buffer can't grow beyond the final counts?

Reply:
Yes, the intention is that there are mainly two use cases for Geometry:
1. We know the final count, so we set these `finalIndexCount` and `finalVertexCount` values to resize the vector once at the beginning
2. We don't know the final count so buffers will have to be resized for every vertex/index append

Reply:
if this is the case, would `maxIndexCount` and `maxVertexCount` be better?

Reply:
Done